docs/DeveloperGuide.md

```diff
@@ -6,24 +6,177 @@
 
 ## Design & implementation
```

Comment:
Maybe you could remove the "C" symbol in all your class diagrams to adhere to what is taught in lecture